rasa/core/agent.py

```diff
@@ -389,8 +389,14 @@ def update_model(
 
         # update domain on all instances
         self.tracker_store.domain = domain
+        if hasattr(self.nlg, "responses"):
+            self.nlg.responses = domain.responses if domain else {}
+
         if hasattr(self.nlg, "templates"):
-            self.nlg.templates = domain.templates if domain else {}
+            rasa.shared.utils.io.raise_deprecation_warning(
+                "The terminology 'template' is deprecated and replaced by 'response'. This will be removed in the future."
```

Comment:
let's be explicit about it so we can actually remove it. Please also add this to the changelog
```suggestion
                "The terminology 'template' is deprecated and replaced by 'response'. This will be removed in Rasa Open Source 3.0.0."
```